src/basics.ml

```diff
@@ -242,13 +242,11 @@ let distinct_vars : term array -> bool =
 
 (** [is_pattern t] tells whether term [t] is a valid pattern. *)
 let rec is_pattern : term -> bool = fun t ->
-  match get_args (unfold t) with
-  | (Patt(_), []  ) -> true (* Pattern variables are not applied *)
-  | (Patt(_), args) -> distinct_vars (Array.of_list args)
-  | (Abst(_,b), []) ->
-      let _, b = Bindlib.unbind b in
-      is_pattern b
-  | (_      , args) -> List.for_all is_pattern args
+  match get_args t with
+  | (Patt(_,_,ts), []  ) -> distinct_vars ts
+  | (Patt(_)     , args) -> distinct_vars (Array.of_list args)
```

Comment:
I would have a FIXME after the definition of the function about this. I think we should disallow patterns like `&X x y` in the future.